InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -110,6 +110,17 @@ QList<Optional<TupleSet> > CompositeQuery::execute(QList<TupleSet> input)
 	return outNode_->calculatedOutputs_;
 }
 
+Query* CompositeQuery::addQuery(std::unique_ptr<Query> query)
+{
+	auto rawQ = query.get();
+	// A query should only be added once:
+	Q_ASSERT(std::find_if(nodes_.begin(), nodes_.end(), [rawQ](QueryNode* existing) {return existing->q_.get() == rawQ;})
+					== nodes_.end());
+	auto newNode = new QueryNode(std::move(query));
```

Comment:
Hmm, instead of using `move` here, it's more correct to change the parameter type of the method to `&&`. Then you might need to use `move` in the client code, but that's the legitimate place to do this.